src/main/scala/is/hail/expr/ir/PruneDeadFields.scala

```diff
@@ -687,6 +694,12 @@ object PruneDeadFields {
         val expr2 = rebuild(expr, child2.typ, memo)
         val newKey2 = rebuild(newKey, child2.typ, memo)
         TableKeyByAndAggregate(child2, expr2, newKey2, nPartitions, bufferSize)
+      case LocalizeEntries(child, fieldName) =>
+        val child2 = rebuild(child, memo)
+        if (child2.typ.rvRowType.fields.contains(MatrixType.entriesIdentifier))
```

Comment:
huh, my comment last night must have gotten lost in the crappy train internet.

This is the bug here -- `IndexedSeq[Field].contains(String)` will always return `false` (when I read it earlier, I thought it would always return true, which it would if you replaced `contains` with `hasField`). Should be `dep.rowType.hasField(fieldName)`